crates/namada/src/ledger/native_vp/multitoken.rs

```diff
@@ -58,18 +68,44 @@ where
         let native_token = self.ctx.pre().ctx.get_native_token()?;
         let is_native_token_transferable =
             is_native_token_transferable(&self.ctx.pre())?;
-        // Native token can be transferred to `PoS` or `Gov` even if
-        // `is_native_token_transferable` is false
-        let is_allowed_inc = |token: &Address, target: &Address| -> bool {
+        let actions = self.ctx.read_actions()?;
+        // The native token can be transferred to and out of the `PoS` and `Gov`
+        // accounts, even if `is_native_token_transferable` is false
+        let is_allowed_inc = |token: &Address, bal_owner: &Address| -> bool {
             *token != native_token
                 || is_native_token_transferable
-                || *target == POS
-                || *target == GOV
+                || actions.iter().any(|action| {
```

Comment:
```suggestion
                || actions.iter().all(|action| {
```
might need `all` instead!

Reply:
yeah definitely `all` for both